src/uast.cc

```diff
@@ -38,6 +42,52 @@ struct Nodes {
   int cap;
 };
 
+enum XPathType {
```

Comment:
I don't understand.

Reply:
XPath apparently has 10 possible types because someone (libxml I guess), has decided so. Yet we're using our own type instead of using libxml's one, so it would be nice to document where those types come from. Something like:

```
// Possible result types for XPath as defined in http://wherever.is/defined
```

Reply:
Ah ok, I understand now, I'll update.

Reply:
I understand now, thanks.

Actually, now that I'm sure `UastFilterTyped` won't be publicly I'll just use the libxml2 enum and remove this one.